qiskit/circuit/library/fourier_checking.py

```diff
@@ -84,7 +84,7 @@ def __init__(self,
                                "tables, each as a list of 2**n entries of "
                                "{1, -1}.")
 
-        super().__init__(num_qubits, name="fc: %s, %s" % (f, g))
+        super().__init__(int(num_qubits), name="fc: %s, %s" % (f, g))
```

Comment:
This will be casted in the `QuantumCircuit` already. Is there any scenario where is better to capture this here?